### DISCUSSION TITLE
Enable AUPRO  (Per Region Overlap) metric to support mask

### DISCUSSION THREAD

**Discussion by triet1102** [open-edge-platform/anomalib#974] [Feature Requests]
Hello, I am currently using AUPRO to evaluate anomaly localization performance and it seems to be that we are not able to filter out points when calculate the metric?

I am asking if we can add support for masking to filter out unnecessary points?
A workaround that I am doing is 
* create a wrapper class for AUPRO
* override the init class which adds inspection_mask parameter
* filter out unnecessary points right after the connected component analysis https://github.com/openvinotoolkit/anomalib/blob/c92fb520d25911c5864ee251e4492348b7cc66c7/src/anomalib/utils/metrics/aupro.py#L87-L94

Thanks in advance

**Comment by ORippler** [comment]
I guess you could split/refactor the `_compute` function into two separate `perform_cca` and `compute_pro` functions. This would give you the ability to modify `preds/cca/target` based on your masking without forcing the need to support and test this in anomalib.

**Reply by triet1102** [reply]
OK thanks @ORippler , I'll wait for the verdict before proposing a PR for that.

**Reply by ORippler** [reply]
CC @samet-akcay before opening a PR

**Reply by samet-akcay** [reply]
Thanks for the reminder @ORippler! I've missed this.

Splitting `_compute` to `perform_cca` and `compute_pro` functions sounds good to me.

**Comment by samet-akcay** [comment]
Closing this as it has already been implemented.
